plugins/controlboard/include/ControlBoardData.hh

```diff
@@ -66,11 +76,15 @@ class ControlBoardData
 {
 public:
     std::mutex mutex;
-    std::vector<JointProperties> joints;
+    std::vector<PhysicalJointProperties> physicalJoints;
+    std::vector<ActuatedAxesProperties>  actuatedAxes;
     yarp::os::Stamp simTime;
-
+    yarp::dev::IJointCoupling* ijointcoupling{nullptr};
```

Comment:
Can we try to use a smart pointer like `std::shared_ptr` instead of raw pointers?

Reply:
mmm I am not a great fan of using smart pointers w/ YARP interfaces, because the memory ownership when you do the `view` is not very clear to me

Reply:
Yes, I can confirm. If you need to access a pointer that is not managed as smart point, you can't use smart pointers.

Reply:
Ah ok thanks understood now!